File: examples/Delphi/Simple/ObjectSpace/DataSetAndReporting/FfrmReport.pas
unit FfrmReport;

interface

uses
  Classes,
  Graphics,
  Controls,
  Forms,
  QuickRpt,
  Qrctrls,
  ExtCtrls,
  Db,
  BoldDataSet,
  BoldSubscription,
  BoldHandles,
  BoldRootedHandles,
  BoldAbstractListHandle,
  BoldCursorHandle,
  BoldListHandle;

type
  TfrmReport = class(TForm)
    qrMain: TQuickRep;
    bndTitle: TQRBand;
    QRLabel1: TQRLabel;
    QRLabel2: TQRLabel;
    blhBuildings: TBoldListHandle;
    blhResidents: TBoldListHandle;
    bdsResident: TBoldDataSet;
    bdsBuilding: TBoldDataSet;
    bndResidents: TQRSubDetail;
    QRDBText3: TQRDBText;
    QRDBText4: TQRDBText;
    bndBuildings: TQRSubDetail;
    QRDBText1: TQRDBText;
    QRDBText2: TQRDBText;
    bndResidentsHeader: TQRBand;
    QRLabel3: TQRLabel;
    QRLabel4: TQRLabel;
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmReport: TfrmReport;

implementation

uses fMain, dMain;

{$R *.DFM}

end.

File: examples/CBuilder/Simple/ObjectSpace/DataSetAndReporting/FfrmReport.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "FfrmReport.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BoldAbstractListHandle"
#pragma link "BoldCursorHandle"
#pragma link "BoldDataSet"
#pragma link "BoldHandles"
#pragma link "BoldListHandle"
#pragma link "BoldRootedHandles"
#pragma link "BoldSubscription"
#pragma resource "*.dfm"
TfrmReport *frmReport;
//---------------------------------------------------------------------------
__fastcall TfrmReport::TfrmReport(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
